src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -241,6 +238,13 @@ public async Task SaveAsync()
 
                 this.TipHashHeight = this.provenBlockHeaderRepository.TipHashHeight;
             }
+
+            if (this.initialBlockDownloadState.IsInitialBlockDownload())
+            {
+                // During IBD the PH cache is not used much,
+                // to avoid occupying unused space in memory we flush the cache.
+                this.Cache.FlushCache();
```

Comment:
there still will be a race condition. 

items that were added to `pendingBatch` after we've exited the lock on 223 and before `PutAsync` finished execution are expected to be in cache but we will remove them.